c/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+cmake_minimum_required(VERSION 3.9)
+
+project(libblake3 VERSION 0.1.0 DESCRIPTION "BLAKE3 C implementation")
+enable_language(C ASM)
+
+include(GNUInstallDirs)
+
+set(blake3_SOURCES)
+list(APPEND blake3_SOURCES
+  blake3.c
+  blake3_dispatch.c
+  blake3_portable.c)
+
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64)
+  list(APPEND blake3_SOURCES
+    blake3_avx2_x86-64_unix.S
+    blake3_avx512_x86-64_unix.S
+    blake3_sse41_x86-64_unix.S)
+elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL i686)
+  list(APPEND blake3_SOURCES
+    blake3_avx2.c
+    blake3_avx512.c
+    blake3_sse41.c)
+  set_source_files_properties(blake3_avx2.c PROPERTIES COMPILE_FLAGS -mavx2)
+  set_source_files_properties(blake3_avx512.c PROPERTIES COMPILE_FLAGS "-mavx512f -mavx512vl")
+  set_source_files_properties(blake3_sse41.c PROPERTIES COMPILE_FLAGS -msse4.1)
+elseif((ANDROID_ABI STREQUAL armeabi-v7a) OR
+       (CMAKE_SYSTEM_PROCESSOR STREQUAL aarch64) OR (CMAKE_SYSTEM_PROCESSOR STREQUAL arm64))
+  list(APPEND blake3_SOURCES blake3_neon.c)
+  set_source_files_properties(blake3_dispatch.c PROPERTIES COMPILE_FLAGS -DBLAKE3_USE_NEON=1)
+  set_source_files_properties(blake3_neon.c PROPERTIES COMPILE_FLAGS -mfpu=neon)
+endif()
+
+configure_file(libblake3.pc.in libblake3.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libblake3.pc
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+add_library(blake3-shared SHARED ${blake3_SOURCES})
+set_target_properties(blake3-shared PROPERTIES VERSION ${PROJECT_VERSION})
+set_target_properties(blake3-shared PROPERTIES SOVERSION 0)
+set_target_properties(blake3-shared PROPERTIES PUBLIC_HEADER blake3.h)
+target_include_directories(blake3-shared PUBLIC .)
+install(TARGETS blake3-shared
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+add_library(blake3-static STATIC ${blake3_SOURCES})
+set_target_properties(blake3-static PROPERTIES OUTPUT_NAME blake3)
+set_target_properties(blake3-static PROPERTIES VERSION ${PROJECT_VERSION})
+set_target_properties(blake3-static PROPERTIES SOVERSION 0)
+set_target_properties(blake3-static PROPERTIES PUBLIC_HEADER blake3.h)
+set_target_properties(blake3-static PROPERTIES COMPILE_FLAGS -fPIC)
```

Comment:
Flags like `-fPIC` here and the `*_unix.S` files above aren't going to work on Windows. How feasible do you think it might be to get this build working across Linux/macOS/Windows, at least for x86-64? As part of that, it would be great to test this build in CI. Here's a minimal way we might do that, and I'm sure other people who knows CMake/GitHub better than I do can suggest improvements:

```diff
diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index d7d754f..5b3243f 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -233,3 +233,19 @@ jobs:
     - name: build b3sum
       run: cargo build --target aarch64-apple-darwin
       working-directory: ./b3sum
+
+  cmake_build:
+    name: CMake ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "macOS-latest", "windows-latest"]
+    steps:
+    - uses: actions/checkout@v3
+    - name: CMake generate
+      run: cmake .
+      working-directory: ./c
+    - name: CMake build
+      run: cmake --build .
+      working-directory: ./c
```

Reply:
I am really not good at CI but I can do it tomorrow morning.

Reply:
My problem is mainly that I'm really not good at CMake :-D I'm pretty sure that snippet of YAML gets the build running, but I'm not sure how to actually fix the errors that come up on the Windows runner.

Reply:
Oh I can edit the YAML script to work, but yeah windows is my main concern too, as of now. I do not have a windows machine around so i'll just make a VM or something tomorrow morning and make sure it works in the next commit. Sorry for the delay btw :)

Reply:
Oh not at all. Thanks for contributing this.

Reply:
and it's done, i'll push right away :)

Reply:
well, no windows machines